### PR TITLE
Project Cluedo - Linda Hintze

### PR DESCRIPTION
**Pull request by lindahz**
**Link to my deployed project:** https://thegameofcluedo.netlify.app/

**My reflections on how this week's project turned out:**
I find JS to be rather complicated. It takes time for me to understand and process all information from the lectures and learning material since I am a slow learner. I hope it will all be clearer in a couple of weeks. 

For this week (and next week) I have and will only reach for the minimum requirements due to stress and busy periods at work. I do feel stressed (also when I see so many people completing the strech goals) in fear of falling behind, but unfortunately I do not have the time right now.. :( 

**Things I'd like to have clarified or explained in more detail:**
Right now I feel like I am behind and I don't know what questions to ask. I have to get up to speed first.

### CODE REVIEW

File: code/clue.js [hunk 1]
```diff
@@ -1,35 +1,265 @@
...
+
+const pistol = {
+  name: 'Pistol',
+  weight: 1,
+  danger: 'High',
+  color: '#FFB7B2',
+  image: 'weapons/pistol.png'
+}
+
+// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE. (I will create objects since I want the rooms to have images).
```
Comment (moablomkvist) [inline code/clue.js:144]: Well done! Images improves the experience of the game.

File: code/clue.js [hunk 2]
```diff
@@ -41,19 +271,73 @@ const randomSelector = array => {
...
+    document.getElementById('mystery').innerHTML = 'Please click on each card before you reveal the crime.'
+    window.scrollBy(0, 300); 
+  } else {
+    document.getElementById('reveal').style.display = 'flex'; 
+    document.getElementById('mystery').innerHTML = `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName} in the ${mystery.room.name} with a ${mystery.weapon.name}.`
+    window.scrollBy(0, 300); 
+  }
+}
+
+// FUNCTION THAT WILL CLEAR THE REVEAL
```
Comment (moablomkvist) [inline code/clue.js:329]: I like that you added the opportunity to clear. Very useful.

File: code/clue.js [hunk 3]
```diff
@@ -41,19 +271,73 @@ const randomSelector = array => {
...
+  roomIcon.remove();
+}
+
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+
+const revealMystery = () => {
+  if (mystery.killer == null || mystery.weapon == null || mystery.room == null) {
+    document.getElementById('reveal').style.display = 'flex'; 
+    document.getElementById('mystery').innerHTML = 'Please click on each card before you reveal the crime.'
```
Comment (moablomkvist) [inline code/clue.js:320]: I do not really understand why I should have to press each card to reveal the crime. 🤔

File: README.md [hunk 4]
```diff
@@ -1,13 +1,17 @@
...
+To create the game I started with creating arrays of objects containing suspects, weapons and rooms. Then I used a math. function to randomly select different suspects, weapons and rooms. The random suspect, weapon and room will show after clicking on each card on the deck, which are all connected to a event handlers that executes a function for each card. To reveal the mystery, I created another event handler that will execute a reveal mystery function. I also created a event handler that will give you the possibility to refresh the page and start a new game.
+
+The code was written in VS Code and in pararell I tested and viewed the site by using devtools.
+
+Before deploying the news site, I tested it on several browsers such as google chrome, safari and firefox. If I had more time I would have made the site responsive. For now it only works on computer devises.
 
 ## View it live
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+A live demo of the website can be found on Netlify: https://thegameofcluedo.netlify.app/
```
Comment (moablomkvist) [inline README.md:24]: Love your design. The color choice feels very appropriate to the theme and I like the clean design.

File: code/clue.js [hunk 5]
```diff
@@ -1,35 +1,265 @@
...
+  firstName: 'Mrs',
+  lastName: 'White',
+  color: '#E7D5A1',
+  description: 'She knows the gossip',
+  age: 62,
+  image: 'assets/white.png',
+  occupation: 'Maid'
+}
+
+// declare the front icons on cards
```
Comment (moablomkvist) [inline code/clue.js:65]: I like comments like this. It makes the code is easy to read and follow which parts belong to what in the code.

File: code/index.html [hunk 6]
```diff
@@ -4,33 +4,60 @@
...
-    <link rel="stylesheet" href="style.css" />
-    <title>Technigo Project Cluedo</title>
+    <link rel="stylesheet" href="style.css"/>
+    <title>A Game of Cluedo</title>
   </head>
   <body>
+
+<!-- HEADER SECTION -->
+    <header>
+      <h1 class="header-title">
```
Comment (moablomkvist) [inline code/index.html:14]: Good  hierarchy in the headings.

File: code/index.html [hunk 7]
```diff
@@ -4,33 +4,60 @@
...
+      </h1>
+      <h2 class="header-subtitle">
+        Start your game by clicking on each card below.
+      </h2>
+    </header>
+
+<!-- CARDS SECTION -->
     <section class="cards">
-      <article class="killer-deck" id="killerCard">
-        <p>KILLER</p>
```
Comment (moablomkvist) [inline code/index.html:13]: This might be an h3 or h4 because the style differs from the other body text, just a thought.

File: code/index.html [hunk 8]
```diff
@@ -4,33 +4,60 @@
...
-        <p>ROOM</p>
+      <article class="room-deck" id="roomCard" onclick='pickRoom()'>
+        <p class="card-title">ROOM</p>
+        <img id="roomIcon" class="deck-image" src="./room.png" alt="room">
         <p id="roomName"></p>
+        <img id="roomImage" class="room-images" src=""/>
       </article>
     </section>
 
-    <h1 id="mystery"></h1>
```
Comment (moablomkvist) [inline code/index.html:31]: One page should only have one h1. Like a book only has one title.

File: code/index.html [hunk 9]
```diff
@@ -4,33 +4,60 @@
...
+  <!-- REVEAL SECTION -->
+    <div class="buttons-wrapper">
+      <button class="button" onclick="revealMystery()">REVEAL THE CRIME</button>
```
Comment (moablomkvist) [inline code/index.html:51]: I really like the idea of ​​the clear-button!!

File: code/style.css [hunk 10]
```diff
@@ -22,37 +69,138 @@ p {
 .killer-deck {
   width: 250px;
   height: 400px;
-  background: #000;
+  background: whitesmoke;
```
Comment (moablomkvist) [inline code/style.css:73]: Good to keep in mind: Be consistent with hex color, since because you've used it everywhere else. :)

**Review by moablomkvist — COMMENTED** [review]
Really good job Linda!
Really nice result with both functionality and design. You should be proud!

**Review by moablomkvist — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (puj) [reply README.md:24]: I think this project looks absolutely fantastic!

File: code/clue.js (hunk 5 above)
Comment (puj) [reply code/clue.js:65]: 100% agree, well done

File: code/clue.js (hunk 1 above)
Comment (puj) [reply code/clue.js:144]: And also creating a `weapons` folder! Very nice.  
Also consider `images/weapons/pistol.png` in a larger project

File: code/clue.js [hunk 11]
```diff
@@ -41,19 +271,73 @@ const randomSelector = array => {
...
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
   document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerName').innerHTML = `Name: ${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerAge').innerHTML = `Age: ${mystery.killer.age}`
+  document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`
+  document.getElementById('killerImage').src = mystery.killer.image
+  document.getElementById('killerDescription').innerHTML = mystery.killer.description
```
Comment (puj) [inline code/clue.js:290]: Excellent that you realized you did not need string templating here

File: code/clue.js [hunk 12]
```diff
@@ -41,19 +271,73 @@ const randomSelector = array => {
...
+  weaponIcon.remove();
+}
+
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms)
+  document.getElementById('roomName').innerHTML = mystery.room.name
+  document.getElementById('roomImage').src = mystery.room.image
+  roomIcon.remove();
+}
+
```
Comment (puj) [inline code/clue.js:313]: Super clean `pickXXX` functions

File: code/clue.js (hunk 3 above)
Comment (puj) [reply code/clue.js:320]: I guess a crime is not complete without a weapon, room and killer?

File: code/clue.js [hunk 13]
```diff
@@ -41,19 +271,73 @@ const randomSelector = array => {
...
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+
+const revealMystery = () => {
+  if (mystery.killer == null || mystery.weapon == null || mystery.room == null) {
```
Comment (puj) [inline code/clue.js:318]: Good logic 👍

File: code/clue.js [hunk 14]
```diff
@@ -41,19 +271,73 @@ const randomSelector = array => {
...
+}
+
+// FUNCTION THAT WILL CLEAR THE REVEAL
+
+const clearRevealResults = () => {
+  mystery.killer = null;
+  mystery.weapon = null;
+  mystery.room = null;
+  location.reload();
+}
```
Comment (puj) [inline code/clue.js:336]: Very nice.  And good function name

File: code/index.html (hunk 6 above)
Comment (puj) [reply code/index.html:14]: Also good class naming!

File: code/style.css (hunk 10 above)
Comment (puj) [reply code/style.css:73]: Indeed, consistency is key.  Hex colors are preferred 👍

File: code/style.css [hunk 15]
```diff
@@ -22,37 +69,138 @@ p {
...
+  width: 150px;
+  padding: 40px 0px;
 }
 
 .room-deck {
   width: 250px;
   height: 400px;
-  background: midnightblue;
-
+  background: whitesmoke;;
```
Comment (puj) [inline code/style.css]: Oops! Double ;

File: code/style.css [hunk 16]
```diff
@@ -22,37 +69,138 @@ p {
...
+}
+
+.deck-image {
+  width: 50%;
+  padding-top: 80px;
+}
+
+.killer-deck:hover, .weapon-deck:hover, .room-deck:hover {
+  opacity: 0.8;
+}
```
Comment (puj) [inline code/style.css:103]: Nice!

**Comment by puj** [comment]
@lindahz You are on an extremely good level for the moment.  You made a very good project. Perhaps you feel like you put less focus on the Javascript portion, but the Javascript you did do was clear, clean, understandable and accurate.  Well done.  You should be proud. 

@moablomkvist Good encouragement and reminders in appropriate places.  Good comments and questions raised 👍

**Review by puj — APPROVED** [review]

**Pull request closed without merging**